server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/MDNSendMethod.scala

```diff
@@ -0,0 +1,316 @@
+import org.apache.james.jmap.mail.MDN._
+import org.apache.james.jmap.mail.MDNSend.MDN_ALREADY_SENT_FLAG
+import org.apache.james.jmap.mail._
+import org.apache.james.jmap.method.EmailSubmissionSetMethod.LOGGER
+import org.apache.james.jmap.routes.{ProcessingContext, SessionSupplier}
+import org.apache.james.lifecycle.api.Startable
+import org.apache.james.mailbox.model.{FetchGroup, MessageResult}
+import org.apache.james.mailbox.{MailboxSession, MessageIdManager}
+import org.apache.james.mdn.fields.{ExtensionField, FinalRecipient, Text}
+import org.apache.james.mdn.{MDN, MDNReport}
+import org.apache.james.metrics.api.MetricFactory
+import org.apache.james.mime4j.codec.DecodeMonitor
+import org.apache.james.mime4j.dom.Message
+import org.apache.james.mime4j.field.AddressListFieldLenientImpl
+import org.apache.james.mime4j.message.DefaultMessageBuilder
+import org.apache.james.mime4j.stream.MimeConfig
+import org.apache.james.queue.api.MailQueueFactory.SPOOL
+import org.apache.james.queue.api.{MailQueue, MailQueueFactory}
+import org.apache.james.server.core.MailImpl
+import play.api.libs.json.{JsError, JsObject, JsSuccess, Json}
+import reactor.core.scala.publisher.{SFlux, SMono}
+import reactor.core.scheduler.Schedulers
+
+import javax.annotation.PreDestroy
+import javax.inject.Inject
+import javax.mail.internet.MimeMessage
+import scala.jdk.CollectionConverters._
+import scala.jdk.OptionConverters._
+import scala.util.Try
+
+class MDNSendMethod @Inject()(serializer: MDNSerializer,
+                              mailQueueFactory: MailQueueFactory[_ <: MailQueue],
+                              messageIdManager: MessageIdManager,
+                              emailSetMethod: EmailSetMethod,
+                              val identifyResolver: IdentifyResolver,
+                              val metricFactory: MetricFactory,
+                              val sessionSupplier: SessionSupplier) extends MethodRequiringAccountId[MDNSendRequest] with Startable {
+  override val methodName: MethodName = MethodName("MDN/send")
+  override val requiredCapabilities: Set[CapabilityIdentifier] = Set(JMAP_MDN, JMAP_MAIL, JMAP_CORE)
+  var queue: MailQueue = _
+
+  def init: Unit =
+    queue = mailQueueFactory.createQueue(SPOOL)
+
+  @PreDestroy def dispose: Unit =
+    Try(queue.close())
+      .recover(e => LOGGER.debug("error closing queue", e))
+
+  override def doProcess(capabilities: Set[CapabilityIdentifier],
+                         invocation: InvocationWithContext,
+                         mailboxSession: MailboxSession,
+                         request: MDNSendRequest): SFlux[InvocationWithContext] = {
+    identifyResolver.resolveIdentityId(request.identityId, mailboxSession)
+      .flatMap(maybeIdentity => if (maybeIdentity.isEmpty) {
+        SMono.raiseError(IdentityIdNotFoundException("The IdentityId cannot be found"))
+      } else {
+        create(maybeIdentity.get, request, mailboxSession, invocation.processingContext)
+      })
```

Comment:
```suggestion
      .flatMap(maybeIdentity =>maybeIdentity.map(identity => create(identity, request, mailboxSession, invocation.processingContext))
          .getOrElse(SMono.raiseError(IdentityIdNotFoundException("The IdentityId cannot be found")))
```

Calling .get is forbidden ;-)